src/main/java/com/sigma/cinematicketpro/service/impl/AuthServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.sigma.cinematicketpro.service.impl;
+
+import com.sigma.cinematicketpro.dto.AuthenticationRequest;
+import com.sigma.cinematicketpro.dto.RegistrationRequest;
+import com.sigma.cinematicketpro.entity.AppUser;
+import com.sigma.cinematicketpro.entity.Role;
+import com.sigma.cinematicketpro.exception.NoSuchRoleException;
+import com.sigma.cinematicketpro.exception.UserAlreadyExistsException;
+import com.sigma.cinematicketpro.repository.UserRepository;
+import com.sigma.cinematicketpro.repository.RoleRepository;
+import com.sigma.cinematicketpro.service.AuthService;
+import com.sigma.cinematicketpro.util.JwtTokenUtils;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.authentication.BadCredentialsException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+@Service
+@RequiredArgsConstructor
+public class AuthServiceImpl implements AuthService {
```

Comment:
you need to have transactions... at least for some operations like registering

Reply:
added transactions to login and registering